### ISSUE TITLE
interleaved understanding, generation video/web data during training

### ISSUE THREAD

**Issue by duyc168** [ByteDance-Seed/Bagel#200]
Thanks for sharing your brilliant work! I have two question here:
1. Could you please clarify the data format in interleaved understanding and interleaved generation (video/web) during training? Are there any concrete data example available? 
2. For interleaved training data, how to compute loss for certain part? for example, the training sample is (image1, text1, image2, text2, image3), I want to compute loss on predicting text2 and image3, how should the code be adapted?

**Comment by Andy1621** [comment]
1. The interleaved data for understanding is like multi-turn converstaions: `Image1 => Text1 => Image2 => Text2`. As for the interleaved generation data, it is in the same data format, and  `UnifiedEditIterableDataset` show the data logic.
2. For understanding data, we only compute loss on text. For generation, we compute loss on image and text. Just set `add_loss=True` in sub_class of `InterleavedBaseIterableDataset` for those parts you want to compute loss.

**Comment by duyc168** [comment]
thanks for the reply! 
I have another issue about the system prompt. I notice that there are system prompts during inference in [inferencer.py](https://github.com/ByteDance-Seed/Bagel/blob/main/inferencer.py) (VLM_THINK_SYSTEM_PROMPT and GEN_THINK_SYSTEM_PROMPT), but I could not find the system prompt that is added during training. Could you tell me where can I find the system prompt for training? Since I want to adapt the system prompt for my task but currently could not find where to adapt it.

**Comment by Andy1621** [comment]
Good question! For the `THINKING_PROMPT`, we modified the dataset, adding the system prompt at the first and insert the thinking prompt in the middle.

We do not provide the dataset in public code, but it's easy to write it~

**Comment by duyc168** [comment]
so for the interleaved generation training data, the order inside the training sample is like [SYSTEM_PROMPT, source image, edit instruction, THINKING_PROMPT, edited image], which is the order during inference? and the SYSTEM_PROMPT is just inherited from Qwen2.5 LLM?

**Comment by Andy1621** [comment]
Please note that we do not add any `SYSTEM_PROMPT` like Qwen. 

The `system prompt` and `THINKING_PROMPT`  I mentioned is the `VLM_THINK_SYSTEM_PROMPT` and `GEN_THINK_SYSTEM_PROMPT`. For thoses dataset with `THINK_SYSTEM_PROMPT`, the order is like:
```
VLM_THINK_SYSTEM_PROMPT => image => instruction => thinking content
GEN_THINK_SYSTEM_PROMPT => source image => edit instruction => thinking content => edited image
```
For inference, just check the inference code.

**Comment by gggggggggguozeyu** [comment]
Thanks for sharing your brilliant work!

Could you please show the structure or an example of the data used for training BAGEL on World Navigation? 

I'm currently planning to fine-tune BAGEL's thinking module with my own dataset. My data is similar to the World Navigation task presented in your paper, except that it includes Chain-of-Thought (CoT) annotations. Specifically, in my dataset, each sample consists of a single input image and an instruction, paired with multiple output images and a corresponding CoT reasoning process.
